accounts/views.py

```diff
@@ -1051,7 +1051,8 @@ def delete(request):
         else:
             delete_sounds =\
                 form.cleaned_data['delete_sounds'] == 'delete_sounds'
-            request.user.profile.delete_user(remove_sounds=delete_sounds)
+            request.user.profile.delete_user(remove_sounds=delete_sounds,
```

Comment:
This will run in the request, which means that if it takes a long time to run, there's a possibility of a timeout.
What options do we have here? If we do it in the background, then the user object will still exist. We could mark password as invalid or mark the account as inactive, and this would prevent the user from logging in. We can always show a message saying "we will process your deletion request and your account/sounds will be deleted shortly". This is especially important if the user has a popular sound and we need to rewrite user download counts... this process could take a long time

Reply:
You're completely right. We should show a message like you say and send the job to gearman to make it work like async user delete in the admin. I'm still considering to do the refactoring of storing delete requests in the DB (as you suggested before) in this PR. If I do that, then it will work using the DB table for deletion requests.

Reply:
Similar fix should be done for sound deletion, but this will go in Sound deletion refactoring PR.

Reply:
Addressed in 7f3e0a8c0b5f7cee8f500fdaf998787f937080e2 (using gearman to delete the user async)